Google.PowerShell/Sql/GcSqlInstanceCmdlets.cs

```diff
@@ -1388,4 +1388,159 @@ private Settings PopulateSetting(Settings newSettings)
+    /// Instance in the Cloud SDK config project. 
+    /// </para>
+    /// <example>
+    ///   <para>Failover the SQL Instance "test1" in the Project "testing."</para>
+    ///   <para><code>PS C:\> Failover-GcSqlReplica -Project "testing" -Instance "test1"</code></para>
+    ///   <br></br>
+    ///   <para>(If successful, the command returns nothing.)</para>
+    /// </example>
+    /// <example>
+    ///   <para>Failover the SQL Instance "test1" with current settings version 3 in the Project "testing."</para>
+    ///   <para>
+    ///     <code>PS C:\> Failover-GcSqlReplica -Project "testing" -Instance "test1" - SettingsVersion 3</code>
+    ///   </para>
+    ///   <br></br>
+    ///   <para>(If successful, the command returns nothing.)</para>
+    /// </example>
+    /// </summary>
+    [Cmdlet("Failover", "GcSqlInstance")]
+    public class FailoverGcSqlInstanceCmdlet : GcSqlCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByName = "ByName";
+            public const string ByInstance = "ByInstance";
+        }
+
+        private class ErrorCodes
+        {
+            public const int InvalidSettingsVersion = 412;
+        }
+
+        private class ErrorMessages
+        {
+            public const string InvalidSettingsVersion =
+                "Input or retrieved settings version does not match current settings version for this instance.";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// Name of the project in which the Instance resides.
+        /// Defaults to the Cloud SDK config for properties if not specified.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByName)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name/ID of the Instance resource to failover.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByName, Mandatory = true, Position = 0)]
+        public string Instance { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The DatabaseInstance that describes the Instance we want to failover.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByInstance, Mandatory = true, Position = 0,
+                   ValueFromPipeline = true)]
+        public DatabaseInstance InstanceObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The current settings version of the Instance. If not specified, it will be retrieved from the settings data
+        /// of the Instance. 
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public long? SettingsVersion { get; set; }
```

Comment:
I could modify the code so we only do a get request if we get only the instance name instead of the instance object. 

The online gui system makes the SettingsVersion required which is a reason I included it as a parameter even though I could just retrieve it manually in all cases. I wasn't sure if perhaps there were situations in which users might only want to failover the instance if it was of a certain settings version or something similar so I kept the specification as an option in both parameter sets.